File: frontend/src/models/Capacity.js
import {Model} from "@/models/Model";

export class Capacity extends Model {
    constructor(id, warehouseId, categoryName, categoryId, capacity, minimumStockLevel) {
        super();
        this.id = id;
        this.warehouseId = warehouseId;
        this.categoryName = categoryName;
        this.categoryId = categoryId;
        this.capacity = capacity;
        this.minimumStockLevel = minimumStockLevel;
    }
}

File: backend/src/main/java/teamx/app/backend/models/Capacity.java
package teamx.app.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import teamx.app.backend.utils.DTO.CapacityDTO;
import teamx.app.backend.utils.Model;

@Data
@Builder
@Entity(name = "WarehouseProductCategoryCapacities")
@NoArgsConstructor
@AllArgsConstructor
public class Capacity implements Model<CapacityDTO>{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private int capacity;
    private int minimumStockLevel;

    @ManyToOne
    @JsonIgnore
    private Warehouse warehouse;

    @ManyToOne
    @JsonIgnore
    private ProductCategory productCategory;

    @Override
    public CapacityDTO toDTO() {
        return CapacityDTO.builder()
                .id(id)
                .capacity(capacity)
                .minimumStockLevel(minimumStockLevel)
                .warehouseId(warehouse != null ? warehouse.getId() : null)
                .categoryId(productCategory != null ? productCategory.getId() : null)
                .categoryName(productCategory != null ? productCategory.getName() : null)
                .build();
    }
}
